src/main/java/uk/gov/hmcts/reform/pip/model/report/AccountMiData.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.reform.pip.model.report;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import uk.gov.hmcts.reform.pip.model.account.Roles;
+import uk.gov.hmcts.reform.pip.model.account.UserProvenances;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Getter
+@Setter
+@NoArgsConstructor
+public class AccountMiData {
+
+    /**
+     * The ID of the user as they exist in P&I.
+     */
+    private UUID userId;
+
+    /**
+     * The user id of the user as per their provenance system.
+     */
+    private String provenanceUserId;
+
+    /**
+     * The Sign in entry system the user was added with. (CFT IDAM, Crime IDAM, P&I AAD).
+     */
+    private UserProvenances userProvenance;
+
+    /**
+     * Role of the user, Verified, Internal or Technical.
+     */
+    private Roles roles;
+
+    /**
+     * The timestamp of when the user was created.
+     */
+    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss'Z'")
+    private LocalDateTime createdDate;
+
+    /**
+     * The timestamp when the user was last signed in.
+     */
+    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss'Z'")
+    private LocalDateTime lastSignedInDate;
+
+    @JsonCreator
+    public AccountMiData(
+        @JsonProperty("userId") UUID userId,
```

Comment:
Why do we need to annotate everything with JsonProperty. It is normally only used when the JSON property name does not match the variable name (for example user_id vs userId). In your case they are all the same. Does it not work without the annotation?

Reply:
Because the formatting method in Publication Services is generic, I had to add the JsonProperty annotations so it can deserialise the responses, rather than needing 4 formatting methods using .get[x] that needed to be kept up-to-date :)